src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1388,6 +1418,273 @@ public async Task TestSchemaIntrospectionQuery(bool enableIntrospection, bool ex
+        [DataRow("/api", HostModeType.Development, false, HttpStatusCode.OK, "{\"urls\":[{\"url\":\"/api/openapi\"", DisplayName = "SwaggerUI enabled in development mode.")]
+        [DataRow("/custompath", HostModeType.Development, false, HttpStatusCode.OK, "{\"urls\":[{\"url\":\"/custompath/openapi\"", DisplayName = "SwaggerUI enabled with custom REST path in development mode.")]
+        [DataRow("/api", HostModeType.Production, true, HttpStatusCode.BadRequest, "", DisplayName = "SwaggerUI disabled in production mode.")]
+        [DataRow("/custompath", HostModeType.Production, true, HttpStatusCode.BadRequest, "", DisplayName = "SwaggerUI disabled in production mode with custom REST path.")]
+        public async Task OpenApi_InteractiveSwaggerUI(
+            string customRestPath,
+            HostModeType hostModeType,
+            bool expectsError,
+            HttpStatusCode expectedStatusCode,
+            string expectedOpenApiTargetContent)
+        {
+            string swaggerEndpoint = "/swagger";
+            Dictionary<GlobalSettingsType, object> settings = new()
+            {
+                { GlobalSettingsType.Host, JsonSerializer.SerializeToElement(new HostGlobalSettings(){ Mode = hostModeType}) },
+                { GlobalSettingsType.GraphQL, JsonSerializer.SerializeToElement(new GraphQLGlobalSettings(){ Enabled = true }) },
+                { GlobalSettingsType.Rest, JsonSerializer.SerializeToElement(new RestGlobalSettings(){ Enabled = true, Path = customRestPath }) }
+            };
+
+            DataSource dataSource = new(DatabaseType.mssql)
+            {
+                ConnectionString = GetConnectionStringFromEnvironmentConfig(environment: TestCategory.MSSQL)
+            };
+
+            RuntimeConfig configuration = InitMinimalRuntimeConfig(globalSettings: settings, dataSource: dataSource);
+            const string CUSTOM_CONFIG = "custom-config.json";
+            File.WriteAllText(
+                CUSTOM_CONFIG,
+                JsonSerializer.Serialize(configuration, RuntimeConfig.SerializerOptions));
+
+            string[] args = new[]
+            {
+                    $"--ConfigFileName={CUSTOM_CONFIG}"
+            };
+
+            using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+            using (HttpClient client = server.CreateClient())
+            {
+                HttpRequestMessage initialRequest = new(HttpMethod.Get, swaggerEndpoint);
+
+                // Adding the following headers simulates an interactive browser request.
+                initialRequest.Headers.Add("user-agent", BROWSER_USER_AGENT_HEADER);
+                initialRequest.Headers.Add("accept", BROWSER_ACCEPT_HEADER);
+
+                HttpResponseMessage response = await client.SendAsync(initialRequest);
+                if (expectsError)
+                {
+                    // Swagger endpoint internally configured to reroute from /swagger to /swagger/index.html
```

Comment:
does this comment make sense in the expectsError section?

Reply:
removed.